composition-js/src/__tests__/override.compose.test.ts

```diff
@@ -0,0 +1,678 @@
+import { printSchema } from "@apollo/federation-internals";
+import gql from "graphql-tag";
+import "./matchers";
+import {
+  assertCompositionSuccess,
+  schemas,
+  errors,
+  composeAsFed2Subgraphs,
+} from "./compose.test";
+
+describe("composition involving @override directive", () => {
```

Comment:
Regarding the test of that file, I'd avoid matching the full supergraph every time. Mostly because if we do, we'll have to update those test every time we have a supergraph format change, even if the change is completely unrelated to those test, and that's annoying. I think we can just have those test check the part of the supergraph they care about, and I've push a commit to do so if you agree: https://github.com/pcmanus/federation/commit/d729eef5da6309b15f9b1c52da3d4e963c25a593.

But another point on testing is that I think we should have least a handful of `@override`-based test in the query planner, because even if the query planner happens to not need modification for this, `@override` is about making sure the query planner doesn't use the overridden fields anymore and we should have tests that test the full feature.